xdsl/dialects/builtin.py

```diff
@@ -344,20 +345,18 @@ def from_list(data: List[_ArrayAttrT]) -> ArrayAttr[_ArrayAttrT]:
 
 
 @irdl_attr_definition
-class DictionaryAttr(GenericData[dict[str, Attribute]]):
+class DictionaryAttr(GenericData[dict[StringAttr, Attribute]]):
```

Comment:
Why do we change this from `str` to `StringAttr`?
We should keep it to `str`.

Reply:
This was my mistake, when I rebased onto a version of main where `validate` checked if the key was an attribute insteaed of val, which confused me greatly. I'll get rid of it!